index.html

```diff
@@ -0,0 +1,261 @@
+              numberCell = row.insertCell(1),
+              booleanCell = row.insertCell(2),
+              dateCell = row.insertCell(3),
+              deleteCell = row.insertCell(4),
+              rowNum = row.rowIndex - 1;
+  
+          tbl[rowNum] = [];
+  
+          addNameCell(nameCell);
+          addNumberCell(numberCell);
+          addBooleanCell(booleanCell);
+          addDateCell(dateCell);
+          addDeleteCell(deleteCell);
+  
+          assignEvents(row);
+  
+          tbl[rowNum].push('');
+          tbl[rowNum].push(Number.MAX_SAFE_INTEGER);
+          tbl[rowNum].push(false);
+          tbl[rowNum].push('1000-01-01');
+      });
+  
+      function sortByColumn() {
+          var i, j, row, cell,
+              col = parseInt(colTypeToNum[this.innerHTML]),
+              table = document.querySelector('#myTable'),
+              tblBody = document.querySelector('#tblBody'),
+              rows = tblBody.rows;
+  
+          if (sortASC[col]) {
+              tbl.sort(function (a, b) {
+                  return (a[col] === b[col]) ? 0 : (a[col] > b[col] ? 1 : -1);
+              });
+          } else {
+              tbl.sort(function (a, b) {
+                  return (a[col] === b[col]) ? 0 : (a[col] > b[col] ? -1 : 1);
+              });
+          }
+  
+          tbl.forEach(function (item, index) {
+              rows[index].cells[0].childNodes[0].value = item[0];
+              rows[index].cells[1].childNodes[0].value = item[1] !== Number.MAX_SAFE_INTEGER ? item[1] : '';
+              rows[index].cells[2].childNodes[0].checked = item[2];
+              rows[index].cells[3].childNodes[0].value = item[3] !== '1000-01-01' ? item[3] : '';
+  
+          });
+          sortASC[col] = !sortASC[col];
+      }
+  
+      function addStringFieldEvent(row) {
+          row.cells[0].childNodes[0].addEventListener('input', function (event) {
+              var input = event.target,
+                  value = input.value,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1;
+  
+              tbl[rowNum][0] = value;
+          });
+      }
+      function addNumberFieldEvent(row) {
+          row.cells[1].childNodes[0].addEventListener('keypress', function (event) {
+              var charCode = (event.which) ? event.which : event.keyCode;
+              if (charCode < 48 || charCode > 57) {
+                  event.preventDefault();
+              }
+          });
+          row.cells[1].childNodes[0].addEventListener('change', function (event) {
+              var input = event.target,
+                  value = input.value,
+                  cell = input.parentNode,
+                  rowNum = cell.parentNode.rowIndex - 1;
+              if (value === '') {
+                  value = Number.MAX_SAFE_INTEGER;
+              }
+              tbl[rowNum][1] = parseInt(value);
+          });
+      }
+      function addBooleanFieldEvent(row) {
```

Comment:
I think every cell should have its own event listener function, But they did got similar... I can do some abstraction.